### PR TITLE
WIP: Sign staging images using cosign

### PR DESCRIPTION
**Pull request by saschagrunert**
#### What type of PR is this?

/kind feature

#### What this PR does / why we need it:
This should make keyless signing into the staging repositories work.
#### Which issue(s) this PR fixes:

None
#### Does this PR have test?
None

#### Special notes for your reviewer:
None
#### Does this PR introduce a user-facing change?

```release-note
None
```

### CODE REVIEW

**Comment by saschagrunert** [comment]
Getting closer, but still have no permissions:

```
> gcloud builds submit --verbosity info --config ./cloudbuild.yaml --project k8s-staging-sp-operator --gcs-log-dir gs://k8s-staging-sp-operator/logs --gcs-source-staging-dir gs://k8s-staging-sp-operator/source
…
+ cosign sign gcr.io/k8s-staging-sp-operator/security-profiles-operator-amd64:12345
Generating ephemeral keys...
Error: signing [gcr.io/k8s-staging-sp-operator/security-profiles-operator-amd64:12345]: getting signer: getting key from Fulcio: fetching ambient OIDC credentials: impersonate: status code 403: {
  "error": {
    "code": 403,
    "message": "The caller does not have permission",
    "status": "PERMISSION_DENIED"
  }
}

main.go:52: error during command execution: signing [gcr.io/k8s-staging-sp-operator/security-profiles-operator-amd64:12345]: getting signer: getting key from Fulcio: fetching ambient OIDC credentials: impersonate: status code 403: {
  "error": {
    "code": 403,
    "message": "The caller does not have permission",
    "status": "PERMISSION_DENIED"
  }
}
make: *** [Makefile:176: image-cross] Error 1
ERROR
```

File: cloudbuild.yaml [hunk 1]
```diff
@@ -5,11 +5,12 @@ options:
   substitution_option: ALLOW_LOOSE
   machineType: N1_HIGHCPU_8
 steps:
-  - name: gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20211118-2f2d816b90
+  - name: gcr.io/k8s-staging-test-infra/gcb-docker-gcloud:v20220506-774f302a94
     entrypoint: bash
     env:
       - TAG=$_GIT_TAG
       - HOME=/root  # for docker buildx
+      - GOOGLE_SERVICE_ACCOUNT_NAME=448637284062@cloudbuild.gserviceaccount.com
```
Comment (JAORMX) [inline cloudbuild.yaml:13]: Mind adding a comment on where this service account name comes from?

File: hack/image-cross.sh [hunk 2]
```diff
@@ -46,6 +56,7 @@ for ARCH in "${ARCHES[@]}"; do
         .
     for T in "${TAGS[@]}"; do
         docker push "$IMAGE-$ARCH:$T"
+        cosign sign "$IMAGE-$ARCH:$T"
```
Comment (JAORMX) [inline hack/image-cross.sh:59]: Will we use cosign's keyless signing? I think you need to provide a reference to the OIDC issuer as a command-line parameter.

File: hack/image-cross.sh [hunk 3]
```diff
@@ -61,6 +72,7 @@ for T in "${TAGS[@]}"; do
     done
 
     docker manifest push --purge "$IMAGE:$T"
+    cosign sign "$IMAGE:$T"
```
Comment (JAORMX) [inline hack/image-cross.sh:75]: We might need the issuer here.

File: hack/image-cross.sh [hunk 4]
```diff
@@ -93,5 +104,6 @@ for I in "${IMAGES[@]}"; do
     for T in "${TAGS[@]}"; do
         docker tag "$I:$VERSION" "$I:$T"
         docker push "$I:$T"
+        cosign sign "$I:$T"
```
Comment (JAORMX) [inline hack/image-cross.sh:107]: neat!! You're signing the bundle too!

**Review by JAORMX — CHANGES_REQUESTED** [review]

File: cloudbuild.yaml (hunk 1 above)
Comment (saschagrunert) [reply cloudbuild.yaml]: Done

**Pull request closed without merging**